java/src/main/native/src/ColumnViewJni.cpp

```diff
@@ -1793,4 +1795,24 @@ JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_ColumnView_copyColumnViewToCV(JNIEnv
   }
   CATCH_STD(env, 0)
 }
+
+JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_ColumnView_getJSONObject(JNIEnv *env, jobject j_object, 
+                                                                     jlong j_view_handle, jlong j_scalar_handle) {
+
+   JNI_NULL_CHECK(env, j_view_handle, "view cannot be null", 0);
+   JNI_NULL_CHECK(env, j_scalar_handle, "path cannot be null", 0);
+
+  try {
+
```

Comment:
All JNI methods that touch the GPU must check if the device needs to be set in case this is a new thread.
```suggestion
    cudf::jni::auto_set_device(env);
```

Reply:
@razajafri